tests/src/tracing/eventcounter/incrementingpollingcounter.cs

```diff
@@ -0,0 +1,138 @@
+        {
+            private object _failureCounter;
+
+            public SimpleEventSource(Func<float> getFailureCount, Type IncrementingPollingCounterType)
+            {
+                _failureCounter = Activator.CreateInstance(IncrementingPollingCounterType, "failureCount", this, getFailureCount);    
+            }
+        }
+
+        internal sealed class SimpleEventListener : EventListener
+        {
+            private readonly string _targetSourceName;
+            private readonly EventLevel _level;
+            private Dictionary<string, string> args;
+            
+            public int FailureEventCount { get; private set; } = 0;
+            public bool Failed = false;
+            public bool MetadataSet = false;
+
+            public SimpleEventListener(string targetSourceName, EventLevel level)
+            {
+                // Store the arguments
+                _targetSourceName = targetSourceName;
+                _level = level;
+                args = new Dictionary<string, string>();
+                args.Add("EventCounterIntervalSec", "1");
+            }
+            
+            protected override void OnEventSourceCreated(EventSource source)
+            {
+                if (source.Name.Equals(_targetSourceName))
+                {
+                    EnableEvents(source, _level, (EventKeywords)(-1), args);
+                }
+            }
+
+            protected override void OnEventWritten(EventWrittenEventArgs eventData)
+            {
+                if (eventData.EventName.Equals("EventCounters"))
+                {
+                    for (int i = 0; i < eventData.Payload.Count; i++)
+                    {
+
+                        // Decode the payload
+                        IDictionary<string, object> eventPayload = eventData.Payload[i] as IDictionary<string, object>;
+
+                        string name = "";
+                        string increment = "";
+
+                        foreach (KeyValuePair<string, object> payload in eventPayload)
+                        {
+                            if (payload.Key.Equals("Name"))
+                            {
+                                name = payload.Value.ToString();
+                            }
+                            else if (payload.Key.Equals("Increment"))
+                            {
+                                increment = payload.Value.ToString();
+                            }
+                        }
+
+                        // Check if the mean is what we expect it to be
+                        if (!increment.Equals("1")) // Increment should always be 1
+                        {
+                            Console.WriteLine($"Incorrect increment: {increment}.");
+                            Failed = true;
+                        }
+                    }
+                }
+            }
+        }
+
+
+        public static int failureCountCalled = 0;
```

Comment:
Can we change this from 'failure' to some other term ( ie _mockedCountCalled)? When "failure" shows up in a test case and it has a non-zero value, the kneejerk interpretation is that the test failed. In this case 'failure' appears to be a stand-in for 'some arbitrary thing we are pretending to count'

Reply:
Ah, didn't consider that part. I'll follow up with another PR to fix that.